File: DigitalPet/app/src/main/java/thedigialex/digitalpet/model/entities/UserDigitalMonster.kt
package thedigialex.digitalpet.model.entities

data class UserDigitalMonster(
    val id: Int,
    val userId: Long,
    val isMain: Int,
    var name: String,
    val type: String,
    val attack: Int,
    val colosseum: Int,
    val level: Int,
    val exp: Int,
    var strength: Int,
    var agility: Int,
    var defense: Int,
    var mind: Int,
    var hunger: Int,
    var exercise: Int,
    var clean: Int,
    var energy: Int,
    val maxEnergy: Int,
    val wins: Int,
    val losses: Int,
    var trainings: Int,
    var maxTrainings: Int,
    var currentEvoPoints: Int,
    var sleepStartedAt: String?,
    val digital_monster: DigitalMonster,
)

File: digital-monster/app/Models/UserDigitalMonster.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class UserDigitalMonster extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'digital_monster_id',
        'isMain',
        'name',
        'type',
        'level',
        'exp',
        'strength',
        'agility',
        'defense',
        'mind',
        'hunger',
        'exercise',
        'clean',
        'energy',
        'maxEnergy',
        'wins',
        'losses',
        'trainings',
        'maxTrainings',
        'currentEvoPoints',
        'sleepStartedAt', 
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function digitalMonster()
    {
        return $this->belongsTo(DigitalMonster::class);
    }

    public function evolve()
    {
        $digitalMonster = $this->digitalMonster;
        if ($this->currentEvoPoints >= $digitalMonster->required_evo_points) {
            $evolutionRoutes = EvolutionRoute::where('evolves_from', $digitalMonster->id)->first();
            if ($evolutionRoutes) {
                if ($this->strength >= $this->mind) {
                    $newMonsterId = $evolutionRoutes->route_a;
                } else {
                    $newMonsterId = $evolutionRoutes->route_b ?? $evolutionRoutes->route_a; 
                }
                $this->digital_monster_id = $newMonsterId;
                $this->currentEvoPoints = 0;
                $this->strength += $this->trainings / 4;
                $this->strength += $this->trainings / 4;
                $this->agility += $this->trainings / 4;
                $this->mind += $this->trainings / 4;
                $this->trainings = 0;
                $this->maxTrainings = ( $this->maxTrainings + 15 ) * 2;
                $this->maxEnergy = $this->maxEnergy + 5;
                $this->energy = $this->maxEnergy;
                $this->save();
                return $this;
            }
        }
        return null;
    }
}
